larq_compute_engine/core/bconv2d_output_transform.h

```diff
@@ -0,0 +1,124 @@
+#ifndef COMPUTE_ENGINE_CORE_OUTPUT_TRANSFORM_H_
+#define COMPUTE_ENGINE_CORE_OUTPUT_TRANSFORM_H_
+
+#include <algorithm>
+#include <cstdint>
+#include <limits>
+
+#include "tensorflow/lite/kernels/internal/common.h"
+#include "tensorflow/lite/kernels/internal/round.h"
+
+namespace compute_engine {
+namespace core {
+
+// Parameters that are needed for both float and int8 kernels
+template <typename AccumScalar>
+struct OutputTransformBase {
+  AccumScalar backtransform_add = 0;
+  AccumScalar clamp_min = std::numeric_limits<AccumScalar>::lowest();
+  AccumScalar clamp_max = std::numeric_limits<AccumScalar>::max();
+
+  AccumScalar Run(const AccumScalar accum) const {
+    // Backtransform can still be done in int32
+    AccumScalar x = backtransform_add - 2 * accum;
+    // Activation function can also be done in int32
+    x = std::min<AccumScalar>(x, clamp_max);
+    x = std::max<AccumScalar>(x, clamp_min);
+    return x;
+  }
+};
+
+template <typename AccumScalar, typename ActivationScalar>
+struct OutputTransform {};
+
+// Parameters that are needed only for float kernels
+template <typename AccumScalar>
+struct OutputTransform<AccumScalar, float> : OutputTransformBase<AccumScalar> {
+  const float* post_activation_multiplier = nullptr;
```

Comment:
In order to add a constructor here, it would require access to several things in `kernels/bconv2d.cc` such as `vector<>`s in the bconv2dparams struct. Therefore I think its cleaner to leave the construction code back in `bconv2d.cc` . What do you think?